yazi-core/src/tab/commands/copy.rs

```diff
@@ -23,13 +28,31 @@ impl Tab {
 		let mut s = OsString::new();
 		let mut it = self.selected_or_hovered(true).peekable();
 		while let Some(u) = it.next() {
-			s.push(match opt.type_.as_str() {
-				"path" => u.as_os_str(),
-				"dirname" => u.parent().map_or(OsStr::new(""), |p| p.as_os_str()),
-				"filename" => u.name(),
-				"name_without_ext" => u.file_stem().unwrap_or(OsStr::new("")),
+			match opt.type_.as_str() {
+				"path" => {
+					match path_to_os_str(u, opt.separator) {
+						Cow::Borrowed(p) => s.push(p),
+						Cow::Owned(p) => s.push(&p),
+					};
+				}
+				"dirname" => {
+					if let Some(parent) = u.parent() {
+						match path_to_os_str(parent, opt.separator) {
+							Cow::Borrowed(p) => s.push(p),
+							Cow::Owned(p) => s.push(&p),
+						};
+					}
+				}
+				"filename" => {
+					s.push(u.name());
+				}
+				"name_without_ext" => {
+					if let Some(stem) = u.file_stem() {
+						s.push(stem);
+					}
+				}
```

Comment:
I need to use `s.push` inside the match expression to avoid cloning `Cow::Borrowed(p)`. Moving `s.push` outside would require cloning `p` (`Cow::to_owned`) to align types correctly (due to scoping issue with `Cow::Owned(p)`), which defeats the purpose of using a COW string.